### PR TITLE
Environments - Updated UI

### PR DESCRIPTION
**Pull request by huzaifa-d**
## Summary of the pull request

This PR updates the UI of the Environments page as per updated requirements and to bring it up to parity with the Setup page.

## Detailed description of the pull request / Additional comments

- Added sorting by name (desc.), alt name (desc.) and last connected (asc.).
- Added Last Sync timer, with one minute, five minute and one-hour intervals. _(The timer starts with showing "Moments ago" when less than a minute has passed, "X minutes ago" for 1-5 minutes in one minute intervals, "X minutes ago" for 10-55 minutes in 5 minute intervals, and "More than an hour ago" for any time after)_
- Added sorting by Provider.
- Fixed sort fields not defaulting back to original values after Sync.
- Fixed resizing which could cause UI elements to disappear.
- Fixed missing glyphs.
- Changed card style to Primary, Tertiary & Secondary brushes.
- Moved the Sync button lower.

Before:
image
After:
image

## Validation steps performed
Manually tested to check if the UI looks as expected

### CODE REVIEW

**Comment by krschau** [comment]
Can you include a screenshot in the PR description?

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (krschau) [inline tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:114]: I wonder if translation is going to not like this. Isn't Ascending/Descending more commonly used?

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (krschau) [inline tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:122]: Because you have this label separate from being the "Header" of the ComboBox, you're eventually going to have an accessibility bug that the ComboBox doesn't have an accessible name. Consider adding it before you do. (Probably easier to do that than re-template the ComboBox to move the Header to the side)

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (krschau) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:60]: Needs to be localized.

Comment (krschau) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:60]: And all the other strings in this file.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (krschau) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:28]: I fixed some of the UI (alignment, accessibility bugs, scrollbar) in this file in PR https://github.com/microsoft/devhome/pull/2423. Keep an eye out for that.
This hard-coded 33 is almost certainly going to cause an accessibility bug regarding text scaling.

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:122]: I didn't quite get the suggestion. Could you elaborate? Should we have 'Sort By'?

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:114]: This is what the designers wanted. I agree, ascending/descending would be better. Since we want parity with the Setup pages, @bbonaby if you don't have any reservations, let's both change this.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:60]: Noted. Will change.

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (krschau) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:122]: Sorry, this was vague out of context, it's just fresh in my mind since I recently fixed #2319. UIA uses the Header property on a ComboBox for its AccessibleName. If you don't have a Header, you'll have to specify an accessible name. I think the easiest way to do it is adding this to your resw:
```
  <data name="SortSelectionComboBox.[using:Microsoft.UI.Xaml.Automation]AutomationProperties.Name" xml:space="preserve">
    <value>Sort by</value>
    <comment>Name of ComboBox where user can choose how to sort</comment>
  </data>
```

You can look at the Environments page in AccessibilityInsights for more context.

Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:122]: Thanks for this. Wouldn't have been able to figure this out on my own.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml:28]: I checked the PR changes and it doesn't seem to affect the Environments page.

Comment (krschau) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml:28]: https://github.com/microsoft/devhome/pull/2423/files#diff-0b6bc2bd8b0085eae1acba1278f5aca975ed022bdfa5551962a34c9f5fb9cbb3

Comment (krschau) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml:28]: I had run it by Branden if I could make these changes since I was already in here fixing the scrolls. This is the bug you'll get if you hard-code the numbers, top is auto and bottom is 37:
image

File: tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs:48]: should we remove the comment? Or are you planning on updating this before build?

File: tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs:89]: We should make an issue on GitHub about adding localization to the operation text

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:187]: Why not say exactly the amount of seconds? Is it a design requirement? Just curious

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:160]: `TimeSpan.FromMinutes(5),` Might be a missed copy/paste from above but seems like you meant to use hours here and not minutes here right?

Also I believe we haven't covered the case for showing "Hours Ago"  and the Day/Days case as well

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:262]: We probably should make sure `All providers` stays at the top of the list regardless of localization

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:270]: Seems a bit weird to do Contains instead of equals. Though I know in the grand scheme of things we don't expect one provider's name to be a subset of another providers name. Using equals ensures we're only showing ones which the provider name only

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:285]: we can remove the comment now

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:169]: This still needs a Todo?

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:286]: Do you know if the Sort options should be sorted in any particular order? E.g ascending or do we not care? If we don't care then this works, but if we do care then we'd need to rework this to look at the selected value instead of selected index. I believe if we wanted it sorted in another language the indexes could possibly change. You'd probably have to create a list of localized strings then use the list as the data source to the combo box. 

If we don't care that the sort options aren't sorted then we're good here

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs]: Moved it.

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (krschau) [inline tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:182]: This should have a {0} placeholder for the number of minutes. It's possible that in another language, the number wouldn't go at the front of this sentence and the translation would be wrong.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:122]: I like the idea, but we didn't need additional time. Might change if the designers come back with wanting more time intervals.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml:169]: I mistook this to be someone else's todo. Will remove in a later PR.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs]: The idea was to not care if they have a longer provider name. But equals would be better.
Changed.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:262]: Yeah, it is made sure of in earlier code.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:154]: No, I meant to use minutes. Updated the description to reflect the current design.

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:187]: Updated the description to reflect the current design.

File: tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs:48]: This is in place so that we don't have to search for the Glyphs again. I don't plan on removing this.

File: tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs:89]: Noted.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:285]: They should be sorted, but I don't understand the localization issue here. Why would the order change? Any localization change would only change texts, not their order.

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw]: TIL.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml]: Removed.

File: tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs
Comment (bbonaby) [reply tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs:38]: I meant the top of it has a ToDo statement so I was under the impression that it was resolved. If the todo is still active we should probably make a github issue. Anis will track them that way.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:256]: Oh, just realized you're not sorting them in the combo box, should they be sorted by name?

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (bbonaby) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:279]: What I meant was right now we have it as 

SortByName
SortByNameDesc
SortByAltName
etc...

in the UI this looks like:

````
Name: Descending
Name: Ascending
Alternative Name: Ascending
````

In english, that isn't sorted. `Alternative Name: Ascending` would show first. In french I'm not sure if they would have the same ordering either. If we hard code the order then sure the index wouldn't change, but then that just means the text is not sorted. So I just wanted to confirm if we cared or not that the text we display to the user wouldn't be alphabetically sorted.

Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:279]: That makes more sense. Yeah, we don't care about that sorting.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (krschau) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:184]: Then if you want the margin above and below you can put it back here: 
```suggestion
                       Margin="0 12 3 12" />
```
(Or do something different, but this gets you something that looks nice quickly)

**Review by krschau — CHANGES_REQUESTED** [review]

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (dhoehna) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:256]: Please use `.Equals` to compare strings.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (dhoehna) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:134]: NIT: Remove `x:Name` if not needed.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (dhoehna) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:145]: Looks like many of your `margins` ae to apply spacing between items.  Consider using the [spacing property](https://learn.microsoft.com/uwp/api/windows.ui.xaml.controls.stackpanel.spacing?view=winrt-22621)

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (dhoehna) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:185]: `Mode=TwoWay` is used when a user can input something.  This is on a `TextBlock` I don't believe users can interact with a `TextBlock`

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml]: Got mixed up due to the merge conflict. Fixed.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs]: Changed.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml]: Needed three lines below.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml:142]: But that would only give me fixed spacing for all the elements.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Views/LandingPage.xaml]: Yes, should be OneWay. Fixed.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:279]: Will keep as-is for this PR.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml
Comment (krschau) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:184]: I got this one slightly wrong -- the StackPanel sees HorizontalAlignment=Right and wants to stick to the right, no matter the available size. What you have right now gives you this when the window is wide:
image

Here's the fix I would use, I tried it out to make sure it works:
```suggestion
        <!-- Last synced text block-->
        <Grid Grid.Row="2" MaxWidth="{ThemeResource MaxPageContentWidth}" Margin="{ThemeResource ContentPageMargin}">
            <StackPanel Orientation="Horizontal" HorizontalAlignment="Right" >
                <TextBlock x:Uid="LastSyncTextBlock"
                           VerticalAlignment="Center"
                           Foreground="{ThemeResource TextFillColorSecondaryBrush}"
                           Margin="0 12 3 12" />
                <TextBlock Text="{x:Bind ViewModel.LastSyncTime, Mode=OneWay}"
                           VerticalAlignment="Center"
                           Foreground="{ThemeResource TextFillColorSecondaryBrush}"/>
            </StackPanel>
        </Grid>
```

**Review by krschau — APPROVED** [review]
One last alignment needs fixing but otherwise approving so my review doesn't hold you up.

**Review by bbonaby — APPROVED** [review]
Feel free to commit Kristen's Change after this approval

**Pull request merged**
